Update image versions in rancher cis benchmark

jakefhyde opened the pull request.

jakefhyde committed:
Basecopy of 0.2.0 version

jakefhyde committed:
Updated chart to use newer security scan and sonobuoy

jiaqiluo commented on charts/rancher-cis-benchmark/0.2.1/Chart.yaml:
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+appVersion: "0.2.0"
+description: |
+  Run CIS benchmark tests
+name: rancher-cis-benchmark
+version: 0.2.0
```
both appVersion and version should be 0.2.1

jakefhyde replied:
Thanks, done.

jiaqiluo reviewed (changes requested):
Why does this PR go to the dev-2.5 branch, not the dev-2.6 branch? Do we want the new version to be only in rancher 2.5.x ?

jakefhyde commented:
> Why does this PR go to the dev-2.5 branch, not the dev-2.6 branch? Do we want the new version to be only in rancher 2.5.x ?

My mistake, I think I was just confused on the difference between system and feature charts. Updated base branch to be dev-v2.6.

cbron commented on charts/rancher-cis-benchmark/0.2.1/Chart.yaml:
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+appVersion: "0.2.1"
```
confirmed this is new dev version

cbron commented on charts/rancher-cis-benchmark/0.2.1/questions.yaml:
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.6.0-alpha1
```
~Should be 2.6.1~ Lets make this 2.6.0, and 0.2.1 should get max of 2.6.0. Do that in this last commit.

cbron commented on charts/rancher-cis-benchmark/0.2.1/values.yaml:
```diff
@@ -0,0 +1,77 @@
+image:
+  securityScan:
+    repository: rancher/security-scan
+    tag: v0.2.4-rc1
```
Need to retag before final RC, sometime next week probably, after this issue is closed.

cbron replied:
Didn't notice before but the last version in 0.2.0 was 0.1.14: https://github.com/rancher/system-charts/blob/release-v2.6/charts/rancher-cis-benchmark/0.2.0/values.yaml#L30
Why didn't we use 0.2.x before in 0.2.0, is that only for v2 ? @prachidamle @paynejacob

prachidamle replied:
Good point @cbron all the changes done in 0.2.0 were for v2. So we have not yet updated the security-scan version for the system-charts since it is used by CIS v1. 

I think most of the changes should be ok, but we need to test CIS v1 before updating the security-scan version. I will run tests and confirm if we are good to update. cc @jakefhyde @paynejacob

prachidamle replied:
I see that the Legacy CIS v1 scan is failing with the updated Sonobuoy versions. We possibly need fixes in rancher CIS v1 controller since it looks for earlier sonobuoy labels ("master" Vs "aggregator"). Will work with @jakefhyde on this

cbron commented on charts/rancher-cis-benchmark/0.2.1/questions.yaml:
```diff
@@ -0,0 +1,2 @@
+rancher_min_version: 2.6.0
+rancher_max_version: 2.6.0
```
This is what the 0.2.0 questions.yaml should be. The 0.2.1 should be just the min.

prachidamle replied:
And I think the min for 0.2.1 should be rancher 2.6.1, correct?

cbron replied:
Technically yes, but it would be ok if you want to make it available in 2.6.0.

paynejacob commented:
@jakefhyde did you run a scan with this new image? There are breaking changes between these versions.

jakefhyde commented:
Environment
-
Install: HA upstream, single node v1.21.4+rke1 downstream (Digital Ocean)

Results
-
|Profile|Passed|Skipped|Failed|Not Applicable|
|---|---|---|---|---|
|RKE-CIS-1.5 Hardened|59|0|13|20|
|RKE-CIS-1.5 Permissive|58|14|0|20|

On first glance, it doesn't look like there is anything anomalous in the scan output.

jakefhyde commented:
Updated pod selector label to `sonobuoy-component=aggregator` and updated label in chart, noted the following error message:

```
time="2021-09-17T04:54:15Z" level=info msg="Running Summarizer" 
time="2021-09-17T04:54:15Z" level=debug msg="version_mapping: map[rke-1.16:rke-cis-1.6-permissive rke-1.19:rke-cis-1.6-permissive eks-1.0:eks-1.0 rke-1.15:rke-cis-1.5-permissive rke-1.17:rke-cis-1.6-permissive gke-1.0:gke-1.0 v1.20.5+rke2r1:rke2-cis-1.6-permissive rke-1.13:rke-cis-1.4 rke-1.14:rke-cis-1.4 rke-1.18:rke-cis-1.6-permissive v1.18.10+rke2r1:rke2-cis-1.5-permissive v1.20.5+k3s1:k3s-cis-1.6-permissive]" 
time="2021-09-17T04:54:15Z" level=info msg="CONFIG: map[v1.20.5+k3s1:k3s-cis-1.6-permissive rke-1.14:rke-cis-1.4 rke-1.18:rke-cis-1.6-permissive v1.18.10+rke2r1:rke2-cis-1.5-permissive rke-1.13:rke-cis-1.4 rke-1.16:rke-cis-1.6-permissive rke-1.19:rke-cis-1.6-permissive eks-1.0:eks-1.0 rke-1.15:rke-cis-1.5-permissive rke-1.17:rke-cis-1.6-permissive gke-1.0:gke-1.0 v1.20.5+rke2r1:rke2-cis-1.6-permissive]\n" 
time="2021-09-17T04:54:15Z" level=info msg="loading checks from config file: /etc/kbs/defaultskip/config.json" 
time="2021-09-17T04:54:15Z" level=info msg="loading checks from config file: /etc/kbs/notapplicable/config.json" 
time="2021-09-17T04:54:15Z" level=error msg="error loading controls from file /etc/kube-bench/cfg/rke-cis-1.5/master.yaml: error reading file /etc/kube-bench/cfg/rke-cis-1.5/master.yaml: open /etc/kube-bench/cfg/rke-cis-1.5/master.yaml: no such file or directory" 
time="2021-09-17T04:54:15Z" level=error msg="error loading controls from file /etc/kube-bench/cfg/rke-cis-1.5/etcd.yaml: error reading file /etc/kube-bench/cfg/rke-cis-1.5/etcd.yaml: open /etc/kube-bench/cfg/rke-cis-1.5/etcd.yaml: no such file or directory" 
time="2021-09-17T04:54:15Z" level=error msg="error loading controls from file /etc/kube-bench/cfg/rke-cis-1.5/node.yaml: error reading file /etc/kube-bench/cfg/rke-cis-1.5/node.yaml: open /etc/kube-bench/cfg/rke-cis-1.5/node.yaml: no such file or directory" time="2021-09-17T04:54:15Z" level=debug msg="total groups loaded: 0" 
time="2021-09-17T04:54:15Z" level=debug msg="total controls loaded: 0" 
time="2021-09-17T04:54:15Z" level=info msg=summarize 
time="2021-09-17T04:54:15Z" level=debug msg="hostDir: jhyde-dev-test-droplet" 
time="2021-09-17T04:54:15Z" level=info msg="found error file" 
time="2021-09-17T04:54:15Z" level=fatal msg="error summarizing: \nThe specified --targets \"etcd\" does not apply to the CIS Benchmark rke-cis-1.5 \\n Valid targets []\n" error running kb-summarizer using override benchmark version
```

cbron commented:
Holding off on this one until later.
